src/main/java/com/rultor/profiles/Profiles.java

```diff
@@ -73,36 +78,108 @@ public Profile fetch(final Talk talk) throws IOException {
             if (xml.nodes("/talk/wire").isEmpty()) {
                 profile = Profile.EMPTY;
             } else {
-                profile = Profiles.fetch(xml);
+                profile = this.fetch(xml);
             }
         }
         return profile;
     }
 
     /**
+     * Merge profile from master and fork. Merged profile must be profile
+     * from fork, but all lists for commanders and architects must be taken
+     * from master branch.
+     * @param master Profile from master branch
+     * @param fork Fork that will be merged
+     * @param branch Fork branch that will be merged
+     * @return Merged profile
+     * @todo #1039:30min Modify method implementation: merged profile is a
+     *  `.rultor.yml` form result of real `git merge` of master branch and
+     *  fork branch. We need real `git merge` result here. See some details
+     *  here: https://github.com/yegor256/rultor/pull/1064/files#r56796959
+     */
+    public Profile merged(final Profile master, final String fork,
+        final String branch) {
+        return master;
+    }
+
+    /**
+     * Validate merged profile: merged profile must have no changes in architect
+     * and commander sections (full match with master profile).
+     * @param master Profile from master branch
+     * @param merged Result of merging master and fork profiles
+     * @return Validated profile
+     * @throws IOException If fails
+     */
+    public Profile validated(final Profile master, final Profile merged)
+        throws IOException {
+        final String commanders = "commanders";
+        final String[][] security = {
+            {"architect"},
+            {Profiles.MERGE, commanders},
+            {"deploy", commanders},
+            {"release", commanders},
+        };
+        for (final String[] section : security) {
+            if (!Arrays.equals(
+                section(master, section),
```

Comment:
@xupyprmv  shouldn't it be `this.section()` here for the method call ? ( we never leave out the `this` :) )